interpolate_phase.py

```diff
@@ -0,0 +1,242 @@
+#!/usr/bin/env python
+
+"""
+Copyright (c) 2023-, California Institute of Technology ("Caltech"). U.S.
+
+Government sponsorship acknowledged.
+All rights reserved.
+Author (s): Emre Havazli
+"""
+
+import argparse
+import glob
+import os
+import shutil
+import time
+
+import matplotlib.pyplot as plt
+import numpy as np
+from osgeo import gdal
+
+EXAMPLE = '''
+interpolate_phase.py <path_to_ifg_folder> <coherence_threshold>
+<interpolation_method>
+'''
+
+
+def create_parser():
+    """Create parser for command line arguments."""
+    parser = argparse.ArgumentParser(
+        description='Interpolate phase',
+        formatter_class=argparse.RawTextHelpFormatter,
+        epilog=EXAMPLE)
+
+    parser.add_argument(
+        'folder', type=str, help='Interferogram folder path')
+
+    parser.add_argument(
+        'coh_thresh', type=float, help='Coherence threshold to select '
+        'reliable pixels')
+
+    parser.add_argument(
+        'interp_method', type=str, help='Interpolation method: chen, chen_v2, numba')
+
+    return parser
+
+
+def get_intf(intf_file):
+    """Get interferogram and coherence from files."""
+    f = intf_file.split('/')
+    f[-1] = 'interp_filt_fine.int'
+    interp_file = os.path.join('/', *f)
+    shutil.copy(intf_file, interp_file)
+    shutil.copy(intf_file + '.xml', interp_file + '.xml')
```

Comment:
I think it may be better to generate a new file instead of copying. Basically adding a save file function and a file name and destination from the parser. Something like this 
`def export_gdal_dataset(outpath, array):
    driver = gdal.GetDriverByName("ENVI")
    dtype = gdal_array.NumericTypeCodeToGDALTypeCode(array.dtype)
    length, width = array.shape
    dset = driver.Create(outpath, xsize=width, ysize=length, bands=1,
                         eType=dtype)
    dset.GetRasterBand(1).WriteArray(array)`